android/src/main/java/app/edge/rnpiratechain/RNPiratechainModule.kt

```diff
@@ -86,12 +86,46 @@ class RNPiratechainModule(private val reactContext: ReactApplicationContext) :
                     }
                 }
             })
+
             wallet.status.collectWith(scope, { status ->
                 sendEvent("StatusEvent") { args ->
                     args.putString("alias", alias)
                     args.putString("name", status.toString())
                 }
             })
+
+            fun handleError(
+                level: String,
+                error: Throwable?,
+            ) {
+                sendEvent("ErrorEvent") { args ->
+                    args.putString("alias", alias)
+                    args.putString("level", level)
+                    args.putString("message", error?.message ?: "Unknown error")
+                }
+            }
+
+            // Error listeners
+            wallet.onCriticalErrorHandler = { error ->
+                handleError("critical", error)
+                false
+            }
+            wallet.onProcessorErrorHandler = { error ->
+                handleError("error", error)
+                true
+            }
+            wallet.onSetupErrorHandler = { error ->
+                handleError("error", error)
+                false
+            }
+            wallet.onSubmissionErrorHandler = { error ->
+                handleError("error", error)
+                false
+            }
```

Comment:
Is `error` always a string?

Reply:
`error` is a `Throwable?`

Reply:
Is a `Throwable?` serializable to a string? I'm wondering if there is a chance the kotlin code could slip some other non-string kind of data type to the JS environment thus making the `message: string` type a lie.